### PR TITLE
FE Updated check-state & FE buyTokens & FE displayTokenBalance

### PR DESCRIPTION
**Pull request by zzuziak**
Creating for @78carla to have a better overview of changes 🙏

### CODE REVIEW

**Comment by 78carla** [comment]
FE Updated check-state & FE buyTokens should. be ready. Paused the displayTokenBalance.

**Comment by zzuziak** [comment]
Hey @78carla just pushed some changes - I'll add individual comments. Main update is fetching the Token address which we can then use to interact with the contract.

I'm now getting a:
`core.mjs:8405 ERROR Error: Uncaught (in promise): TypeError: _this.lotteryTokenContract.balanceOf is not a function`

File: frontend/src/app/app.component.ts [hunk 1]
```diff
@@ -53,6 +58,10 @@ export class AppComponent {
     this.getLotteryContractAddress().subscribe((data) => {
       this.lotteryContractAddress = data.address;
     });
+
+    this.getTokenAddres().subscribe((data) => {
+      this.lotteryTokenContractAddress = data.address;
+    });
```
Comment (zzuziak) [inline frontend/src/app/app.component.ts:64]: Added this bit to retrieve the address of the token contract

File: frontend/src/app/app.component.ts [hunk 2]
```diff
@@ -62,6 +71,23 @@ export class AppComponent {
...
+      lotteryJson.abi,
+      this.userWallet ?? this.provider
+    );
+    const balanceBN = await this.lotteryTokenContract['balanceOf'](
+      this.userWallet
+    );
+
+    const balance = ethers.utils.formatEther(balanceBN);
+    this.userTokenBalance = parseFloat(balance);
   }
```
Comment (zzuziak) [inline frontend/src/app/app.component.ts:91]: And then here, calling the `displayTokenBalance` once a user connects their wallet, but it's failing on line 85 🤔

**Pull request merged**